python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,980 @@
+        dcrModelCut = [model[bbox, afwImage.PARENT] for model in dcrModels]
+        modelSum = afwMath.statisticsStack(dcrModelCut, statsFlags, statsCtrl, modelWeightList,
+                                           clipped,
+                                           maskMap)
+        significanceImage = np.abs(modelSum.image.array)
+        weight = 0
+        metric = 0.
+        metricList = {}
+        zipIterables = zip(tempExpRefList, weightList, imageScalerList, spanSetMaskList)
+        for tempExpRef, expWeight, imageScaler, altMaskSpans in zipIterables:
+            exposure = tempExpRef.get(tempExpName + "_sub", bbox=bbox)
+            imageScaler.scaleMaskedImage(exposure.maskedImage)
+            singleMetric = self.calculateSingleConvergence(exposure, dcrModels, significanceImage, statsCtrl,
+                                                           altMaskSpans=altMaskSpans)
+            metric += singleMetric*expWeight
+            metricList[tempExpRef.dataId["visit"]] = singleMetric
+            weight += expWeight
+        self.log.info("Individual metrics:\n%s", metricList)
+        if weight == 0:
+            return 1.
+        else:
+            return metric/weight
+
+    def calculateSingleConvergence(self, exposure, dcrModels, significanceImage, statsCtrl,
+                                   altMaskSpans=None):
+        """Calculate a quality of fit metric for a single matched template.
+
+        Parameters
+        ----------
+        exposure : lsst.afw.image.exposure.exposure.ExposureF
+            The input warped exposure to evaluate.
+        dcrModels : list of lsst.afw.image.maskedImageF
+            A list of masked images, each containing the model for one subfilter
+        significanceImage : `np.ndarray`
+            Array of weights for each pixel corresponding to its significance
+            for the convergence calculation.
+
+        Returns
+        -------
+        `float`
+            Quality of fit metric for one exposure, within the sub-region.
+        """
+        convergeMask = afwImage.Mask.getPlaneBitMask(self.config.convergenceMaskPlanes)
+        templateImage = self.buildMatchedTemplate(dcrModels, exposure.getInfo().getVisitInfo(),
+                                                  exposure.getBBox(), exposure.getInfo().getWcs())
+        diffVals = np.abs(exposure.image.array - templateImage.image.array)*significanceImage
+        refVals = np.abs(templateImage.image.array)*significanceImage
+
+        finitePixels = np.isfinite(refVals) & np.isfinite(diffVals)
+        if altMaskSpans is not None:
+            self.applyAltMaskPlanes(exposure.mask, altMaskSpans)
+        goodMaskPixels = exposure.mask.array & statsCtrl.getAndMask() == 0
+        convergeMaskPixels = exposure.mask.array & convergeMask > 0
+        usePixels = finitePixels & goodMaskPixels & convergeMaskPixels
+        if np.sum(usePixels) == 0:
+            metric = 0.
```

Comment:
I don't think this does anything, as `metric` is immediately overwritten. Did you mean to include an `else`?